Detector/DetFCChhTrackerSimple/src/GenericEndcap_Geo.cpp

```diff
@@ -0,0 +1,133 @@
+
+#include "DetCommon/DetUtils.h"
+#include "DD4hep/DetFactoryHelper.h"
+
+using DD4hep::Geometry::Volume;
+using DD4hep::Geometry::DetElement;
+using DD4hep::XML::Dimension;
+using DD4hep::Geometry::PlacedVolume;
+
+namespace det {
+/**
+Factory for a configurable, generic tracker endcap.
+@author: Valentin Volkl
+@todo: Fix half-length definitions / factors of two
+*/
+static DD4hep::Geometry::Ref_t createGenericTrackerEndcap(DD4hep::Geometry::LCDD& lcdd,
+                                                          DD4hep::XML::Handle_t xmlElement,
+                                                          DD4hep::Geometry::SensitiveDetector sensDet) {
+  // shorthands
+  DD4hep::XML::DetElement xmlDet = static_cast<DD4hep::XML::DetElement>(xmlElement);
+  Dimension dimensions(xmlDet.dimensions());
+
+  // get sensitive detector type from xml
+  DD4hep::XML::Dimension sdTyp = xmlElement.child("sensitive");  // retrieve the type
+  if (xmlDet.isSensitive()) {
+    sensDet.setType(sdTyp.typeStr());  // set for the whole detector
+  }
+
+  // definition of top volume
+  std::string detName = xmlDet.nameStr();
+  DetElement GenericTrackerEndcapWorld(detName, xmlDet.id());
+
+  // envelope volume with the max dimensions of tracker for visualization etc.
+  // contains both endcaps, in forward and in backwards direction
+
+  DD4hep::Geometry::Tube posnegEnvelopeShape_plus(dimensions.rmin(), dimensions.rmax(), (dimensions.z2()));
+  DD4hep::Geometry::Box posnegEnvelopeShape_minus(
+      dimensions.rmax() * 1.001, dimensions.rmax() * 1.001, dimensions.z1());
```

Comment:
The variable names were a bit misleading, this is actually the envolope for both sides, but I subtract the volume where I assume the Barrel, so there are no envelope-overlaps. I changed the variable names and added a comment.